### ISSUE TITLE
Notification email once user hits 90% of it's quota

### ISSUE THREAD

**Issue by MorrisJobke** [owncloud/core#16258]
Notify the user by email when he hits a given percentage of her or his quota.

How could we do this? Provide the quota usage via API (so external services can hook in here and send emails out). Maybe provide a hook for this, so apps can use it and then do some action?

cc @jancborchardt Do you have an opinion on this UX wise?
cc @nickvergessen @rullzer @DeepDiver1975 @PVince81 Do you have an idea implementation wise?

We could convert this to a junior job once it is ready ;)

**Comment by nickvergessen** [comment]
this should not be a notification email, but a hook. Let activities and others take care of the email stuff.
We could check the quota after upload of a file, but that might be a bit too frequent and at the wrong position (not sure how expensive "getting the quota" is).

One implementation idea:
* When a user uploaded a file, add a timed background job that is run in 1hour or something (so we don't notify on each upload)
* In the background job check the quota and if >= 90% call an implementation of `OCP\Activity\IExtension`

**Comment by DeepDiver1975** [comment]
this is already implemented with web_hooks - https://github.com/owncloud/web_hooks/wiki/Events-and-topics#event-quota

**Comment by nickvergessen** [comment]
Related https://github.com/owncloud/activity/issues/105

**Comment by PVince81** [comment]
@MorrisJobke "quota usage via API": you can get the quota usage of a user by doing a PROPFIND on the webdav root and ask for the properties "{DAV:}quota-available-bytes" and "{DAV:}quota-used-bytes".

But this would require polling. The webhook idea might be better.

**Comment by MorrisJobke** [comment]
> @MorrisJobke "quota usage via API": you can get the quota usage of a user by doing a PROPFIND on the webdav root and ask for the properties "{DAV:}quota-available-bytes" and "{DAV:}quota-used-bytes".

And this requires user credentials ;)

**Comment by MorrisJobke** [comment]
I like that you can configure the barriers for the quota. https://github.com/owncloud/web_hooks/wiki/Events-and-topics#event-quota 

But this is just something for the more technical part of the users. Providing a way to just add a checkmark to get notified once a given threshold is reached is also nice.

Thanks for all the hints. :smile:

**Comment by PVince81** [comment]
Closing in favor of https://github.com/owncloud/core/issues/6878
